Running $ yarn hot gives several type errors

On current DEV, if you run `$ yarn hot`, and wait for the build to finish the webpack and wait a few seconds, you get outputs like this:

```
app/api/auth2fa/routes.ts:24:5 - error TS2345: Argument of type '(req: Request<ParamsDictionary>, res: Response, next: NextFunction) => void | Response' is not assignable to parameter of type 'Middleware'.
  Types of parameters 'req' and 'req' are incompatible.
    Type '{ user: User; body: any; }' is missing the following properties from type 'Request<ParamsDictionary>': get, header, accepts, acceptsCharsets, and 76 more.

24     needsAuthorization(['admin', 'editor']),
       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

app/api/auth2fa/routes.ts:26:12 - error TS7006: Parameter 'req' implicitly has an 'any' type.

26     async (req, res, next) => {
              ~~~

app/api/auth2fa/routes.ts:26:17 - error TS7006: Parameter 'res' implicitly has an 'any' type.

26     async (req, res, next) => {
                   ~~~

app/api/auth2fa/routes.ts:26:22 - error TS7006: Parameter 'next' implicitly has an 'any' type.

26     async (req, res, next) => {
                        ~~~~

app/api/auth2fa/routes.ts:38:5 - error TS2345: Argument of type '(req: Request<ParamsDictionary>, res: Response, next: NextFunction) => void | Response' is not assignable to parameter of type 'Middleware'.

38     needsAuthorization(['admin', 'editor']),
       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

app/api/auth2fa/routes.ts:44:12 - error TS7006: Parameter 'req' implicitly has an 'any' type.

44     async (req, res, next) => {
              ~~~

app/api/auth2fa/routes.ts:44:17 - error TS7006: Parameter 'res' implicitly has an 'any' type.

44     async (req, res, next) => {
                   ~~~

app/api/auth2fa/routes.ts:44:22 - error TS7006: Parameter 'next' implicitly has an 'any' type.

44     async (req, res, next) => {
                        ~~~~

app/api/auth2fa/routes.ts:56:5 - error TS2345: Argument of type '(req: Request<ParamsDictionary>, res: Response, next: NextFunction) => void | Response' is not assignable to parameter of type 'Middleware'.

56     needsAuthorization(['admin']),
       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

app/api/auth2fa/routes.ts:67:12 - error TS7006: Parameter 'req' implicitly has an 'any' type.

67     async (req, res, next) => {
              ~~~

app/api/auth2fa/routes.ts:67:17 - error TS7006: Parameter 'res' implicitly has an 'any' type.

67     async (req, res, next) => {
                   ~~~

app/api/auth2fa/routes.ts:67:22 - error TS7006: Parameter 'next' implicitly has an 'any' type.

67     async (req, res, next) => {
                        ~~~~
```

@daneryl, @konzz or @fnocetti , you are using macOS, can you please confirm this is happening on your end to?

No.  That branch is giving another set of conflicts, more or less of the same nature:

```
app/api/socketio/middleware.ts:1:20 - error TS7016: Could not find a declaration file for module 'cookie'. '/home/rafapolit/workspace/uwazi/node_modules/cookie/index.js' implicitly has an 'any' type.
  Try `npm install @types/cookie` if it exists or add a new declaration (.d.ts) file containing `declare module 'cookie';`

1 import cookie from 'cookie';
                     ~~~~~~~~

app/api/socketio/middleware.ts:2:60 - error TS7016: Could not find a declaration file for module 'socket.io'. '/home/rafapolit/workspace/uwazi/node_modules/socket.io/lib/index.js' implicitly has an 'any' type.
  Try `npm install @types/socket.io` if it exists or add a new declaration (.d.ts) file containing `declare module 'socket.io';`

2 import socketIo, { Server as SocketIoServer, Socket } from 'socket.io';
                                                             ~~~~~~~~~~~
```

--- a/app/api/auth2fa/routes.ts
+++ b/app/api/auth2fa/routes.ts
@@ -1,24 +1,12 @@
 /** @format */
 import Joi from 'joi';
+import { Application } from 'express';
 
 import needsAuthorization from 'api/auth/authMiddleware';
 import * as usersUtils from 'api/auth2fa/usersUtils';
 import { validation } from 'api/utils';
-import { User } from 'api/users/usersModel';
 
-/**
- * This will probably not be required once express types
- * are passed down through a .ts APP or SERVER
- */
-type Middleware = (
-  req: { user: User; body: any },
-  res: { json: any },
-  next: any
-) => Promise<void> | void;
-type Post = { (arg0: string, arg1: Middleware, arg2: Middleware, arg3?: Middleware): void };
-type App = { post: Post };
-
-export default (app: App) => {
+export default (app: Application) => {
   app.post(
     '/api/auth2fa-secret',
     needsAuthorization(['admin', 'editor']),

--- a/app/api/csv/csvLoader.ts
+++ b/app/api/csv/csvLoader.ts
@@ -1,4 +1,4 @@
-import EventEmitter from 'events';
+import { EventEmitter } from 'events';
 
 import templates from 'api/templates';
 import settings from 'api/settings';

